### ISSUE TITLE
[ENH]: Accessibility for SVG hyperlinks

### ISSUE THREAD

**Issue by jonblatho** [matplotlib/matplotlib#21477]
### Problem

Currently matplotlib supports adding hyperlinks in SVGs, but (I believe) it lacks support for setting an `aria-label` attribute on links to improve accessibility.

### Proposed solution

Alongside existing functionality to provide URLs for SVG hyperlinks (like the `url=` arg and `set_urls()`), a user should be able to provide corresponding `aria-label` attributes to be set for `<a>` tags.

### Additional context and prior art

An example use case for this is adding links to an SVG scatter plot for use on the web where the points are too densely packed to make text labeling feasible.
